modules/post/linux/manage/adduser.rb

```diff
@@ -0,0 +1,249 @@
+    fail_with(Failure::NotVulnerable, 'Cannot add user to sudo as sudoers doesnt exist') unless datastore['SudoMethod'] != 'SUDO_FILE' || file_exist?('/etc/sudoers')
+    fail_with(Failure::NotFound, 'Shell specified does not exist on system') unless command_exists?(datastore['SHELL'])
+    fail_with(Failure::BadConfig, "Username [#{datastore['USERNAME']}] is not a legal unix username.") unless datastore['USERNAME'] =~ /^[a-z][a-z0-9_-]{0,31}$/
+
+    # Encrypting password ahead of time
+    passwd = UnixCrypt::MD5.build(datastore['PASSWORD'])
+
+    # Adding sudo to groups if method is set to use groups
+    groups = datastore['GROUPS'].split
+    groups += ['sudo'] if datastore['SudoMethod'] == 'GROUP'
+    groups = groups.uniq
+
+    # Check to see that groups exist or fail
+    group_file = read_file('/etc/group').to_s
+    groups_missing = groups.reject { |group| check_group_exists?(group, group_file) }
+
+    unless groups_missing.empty?
+      if datastore['MissingGroups'] == 'ERROR'
+        fail_with(Failure::NotFound, "groups [#{groups_missing.join(' ')}] do not exist on the system")
+      end
+      print_bad("Groups [#{groups_missing.join(' ')}] do not exist on system")
+      if datastore['MissingGroups'] == 'IGNORE'
+        groups -= groups_missing
+        print_good("Removed #{groups_missing.join(' ')} from target groups")
+      end
+    end
+
+    # Automatically ignore setting groups if added additional groups is empty
+    groups_handled = groups.empty?
+
+    # Check database to see what OS it is. If it meets specific requirements, This can all be done in a single line
+    binary =
+      if datastore['UseraddBinary']
+        datastore['UseraddBinary']
+      elsif command_exists?('useradd')
+        'useradd'
+      elsif command_exists?('adduser')
+        'adduser'
+      else
+        fail_with(Failure::NotFound, 'Cannot find a binary to add a new user cleanly')
+      end
+    os_platform =
+      if session.type == 'meterpreter'
+        sysinfo['OS']
+      elsif active_db? && framework.db.workspace.hosts.where(address: session.session_host) && framework.db.workspace.hosts.where(address: session.session_host).first.os_name
+        host = framework.db.workspace.hosts.where(address: session.session_host).first
+        if host.os_name == 'linux' && host.os_flavor
+          host.os_flavor
+        else
+          host.os_name
+        end
+      else
+        get_sysinfo[:distro]
+      end
+
+    binary_case = if binary =~ /useradd$/
+                    'useradd'
+                  elsif binary =~ /adduser$/
+                    'adduser'
+                  else
+                    binary end
+    case binary_case
+    when 'useradd'
+      print_status("Running on #{os_platform}")
+      print_status('Useradd exists. Using that')
+      case os_platform
+      when /debian|ubuntu|fedora|centos|oracle|redhat|arch|suse|gentoo/i
```

Comment:
I could do that, but it might be better to  keep them in that state for future proofing, say theres another OS that uses useradd that doesnt function the same way as all other implementations. I dont remember if busybox has useradd or if alpine can have useradd on its system. That is something that might be worth digging into